src/ir/module-splitting.cpp

```diff
@@ -228,6 +274,42 @@ ModuleSplitter::classifyFunctions(const Module& primary, const Config& config) {
   return std::make_pair(primaryFuncs, secondaryFuncs);
 }
 
+std::map<Name, Name>
+ModuleSplitter::initExportedPrimaryFuncs(const Module& primary) {
+  std::map<Name, Name> functionExportNames;
+  for (auto& ex : primary.exports) {
+    if (ex->kind == ExternalKind::Function) {
+      functionExportNames[ex->value] = ex->name;
+    }
+  }
+  return functionExportNames;
+}
+
+void ModuleSplitter::exportImportFunction(Name funcName) {
+  Name exportName;
+  // If the function is already exported, use the existing export name.
+  // Otherwise, create a new export for it.
+  auto exportIt = exportedPrimaryFuncs.find(funcName);
+  if (exportIt != exportedPrimaryFuncs.end()) {
+    exportName = exportIt->second;
+  } else {
+    exportName = Names::getValidExportName(
+      primary, config.newExportPrefix + funcName.c_str());
+    primary.addExport(
+      Builder::makeExport(exportName, funcName, ExternalKind::Function));
+    exportedPrimaryFuncs[funcName] = exportName;
+  }
+  // Import the function if it is not already imported into the secondary
+  // module.
```

Comment:
In which case the function can be already imported into the secondary module at this point?

Reply:
For example if `exportImportFunction` has already been called on it. This can happen for a primary function that is called from a secondary function and is also in the table segment with a non-const offset. In that case both `exportImportCalledPrimaryFunctions` and `setupTablePatching` will call `exportImportFunction` on it.